src/include/sof/lib/ams_msg.h

```diff
@@ -0,0 +1,27 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * Author: Ievgen Ganakov <ievgen.ganakov@intel.com>
+ */
+
+#ifndef __SOF_LIB_AMS_MSG_H__
+#define __SOF_LIB_AMS_MSG_H__
+
+/* AMS messages */
+
+/* Key-phrase detected AMS message uuid: 80a11122-b36c-11ed-afa1-0242ac120002*/
+#define AMS_KPD_MSG_UUID 0x80a11122, 0xb36c, 0x11ed, 0xaf, 0xa1, 0x02, 0x42, 0xac, 0x12, 0x00, 0x02
+
+#define DEFINE_UUID_(name, l1, s1, s2, b1, b2, b3, b4, b5, b6, b7, b8) \
+	static const uint32_t name[] = \
```

Comment:
Why `uint32_t` and the endianness and VA_ARGS complexity? This seems to be always cast to `uint8t_t *` in the end anyway.